arch/src/x86_64/smbios.rs

```diff
@@ -134,11 +135,22 @@ pub struct SmbiosEndOfTable {
     pub handle: u16,
 }
 
+#[repr(C)]
+#[repr(packed)]
+#[derive(Default, Copy, Clone)]
+struct SmbiosOemString {
+    typ: u8,
```

Comment:
I would call this `r#type`

Reply:
Do you want to change other structures in the same file too? I can send a follow-up PR to fix them.

Re the Fixes tag, yes I can do that in the future.

Reply:
Oops, sorry I didn't realise it was consistent within the same file. I would prefer it to use the `r#type` consistently. Feel free to push a commit to this PR.

Reply:
Sure no problem. I pushed two more clean-up patches.